tools/c7n_azure/c7n_azure/actions/tagging.py

```diff
@@ -179,16 +157,101 @@ def _process_resource(self, resource, event):
         if not self.should_update and resource.get('tags', {}).get(self.tag_key, None):
             return
 
-        user = self.default_user
+        tag_value = self.default_value
         if event:
-            user = self._get_user_from_event(event) or user
+            tag_value = self._get_tag_value_from_event(event) or tag_value
+        else:
+            tag_value = self._get_tag_value_from_resource(resource) or tag_value
+
+        TagHelper.add_tags(self, resource, {self.tag_key: tag_value})
+
+    def _get_first_event(self, resource):
+
+        if 'c7n:first_iam_event' in resource:
+            return resource['c7n:first_iam_event']
+
+        # Makes patching this easier
+        from c7n_azure.utils import utcnow
+
+        # Calculate start time
+        delta_days = self.data.get('days', self.max_query_days)
+        start_time = utcnow() - datetime.timedelta(days=delta_days)
+
+        # resource group type
+        if self.manager.type == 'resourcegroup':
+            resource_type = "Microsoft.Resources/subscriptions/resourcegroups"
+            query_filter = " and ".join([
+                "eventTimestamp ge '%s'" % start_time,
+                "resourceGroupName eq '%s'" % resource['name'],
+                "eventChannels eq 'Operation'",
+                "resourceType eq '%s'" % resource_type
+            ])
+        # other Azure resources
         else:
-            user = self._get_user_from_resource_logs(resource) or user
+            resource_type = resource['type']
+            query_filter = " and ".join([
+                "eventTimestamp ge '%s'" % start_time,
+                "resourceUri eq '%s'" % resource['id'],
+                "eventChannels eq 'Operation'",
+                "resourceType eq '%s'" % resource_type
+            ])
 
-        # issue tag action to label user
-        TagHelper.add_tags(self, resource, {self.tag_key: user})
+        # fetch activity logs
+        logs = self.client.activity_logs.list(
+            filter=query_filter,
+            select=self.query_select
+        )
+
+        # get the user who issued the first operation
+        operation_name = "%s/write" % resource_type
```

Comment:
I discovered we could potentially be even smarter here. There is a substatus in the activity logs... the problem is I don't think we can guarantee that all RPs use the same text. However, something we can consider investigating.

```
    "subStatus": {
        "value": "Created",
        "localizedValue": "Created (HTTP Status Code: 201)"
    },
```